LEGO1/mxbitmap.cpp

```diff
@@ -28,10 +33,34 @@ int MxBitmap::vtable14(int)
   return 0;
 }
 
-// OFFSET: LEGO1 0x100bcba0 STUB
-int MxBitmap::vtable18(BITMAPINFOHEADER *p_bmiHeader)
+// OFFSET: LEGO1 0x100bcba0
+MxResult MxBitmap::vtable18(BITMAPINFOHEADER *p_bmiHeader)
 {
-  return 0;
+  MxResult result = FAILURE;
+  int width = p_bmiHeader->biWidth;
+  int height = p_bmiHeader->biHeight;
+
+  this->m_info = (BITMAPINFO*) malloc(0x428);
```

Comment:
These hardcoded numbers should probably be avoided. Most likely numbers like `0x428` actually came from somewhere (e.g. `malloc(sizeof(BITMAPINFO))`) so it would be better to figure out their origins . Since this is C++, there's a good chance these might actually be `new`s rather than `malloc`s (e.g. `new BITMAPINFO`), which would figure out the size and malloc automatically.